pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -243,45 +291,68 @@ func (plugin *kubenetNetworkPlugin) Event(name string, details map[string]interf
 		return
 	}
 
-	klog.V(5).Infof("PodCIDR is set to %q", podCIDR)
-	_, cidr, err := net.ParseCIDR(podCIDR)
-	if err == nil {
-		setHairpin := plugin.hairpinMode == kubeletconfig.HairpinVeth
-		// Set bridge address to first address in IPNet
-		cidr.IP[len(cidr.IP)-1] += 1
+	klog.V(4).Infof("kubenet: PodCIDR is set to %q", podCIDR)
+	podCIDRs := strings.Split(podCIDR, ",")
 
-		json := fmt.Sprintf(NET_CONFIG_TEMPLATE, BridgeName, plugin.mtu, network.DefaultInterfaceName, setHairpin, podCIDR, cidr.IP.String())
-		klog.V(2).Infof("CNI network config set to %v", json)
-		plugin.netConfig, err = libcni.ConfFromBytes([]byte(json))
-		if err == nil {
-			klog.V(5).Infof("CNI network config:\n%s", json)
+	// reset to one cidr if dual stack is not enabled
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) && len(podCIDRs) > 1 {
+		klog.V(2).Infof("This node has multiple pod cidrs assigned and dual stack is not enabled. ignoring all except first cidr")
+		podCIDRs = podCIDRs[0:1]
```

Comment:
should we check if each CIDR belongs to a different ip family as Tim suggested previously?